.github/workflows/build-test-virtual.yml

```diff
@@ -5,6 +5,10 @@ on:
       - main
   pull_request:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
Why do we want to ensure that a single job run at a time?

Reply:
This is a single job per-branch at a time. So if you push commit A and later commit B, the build for commit A gets stopped at the one for commit B gets started. It's ok to continue running old builds, but it's wasteful, because even if it passes, it doesn't tell you if the build for commit B will.

Reply:
this